registry/txt.go

```diff
@@ -146,6 +146,11 @@ func (im *TXTRegistry) Records(ctx context.Context) ([]*endpoint.Endpoint, error
 			continue
 		}
 		// We simply assume that TXT records for the registry will always have only one target.
+		// If there are no targets (e.g for routing policy based records in google), direct targets will be empty
+		if len(record.Targets) == 0 {
```

Comment:
this require a unit test with log testing. We have a method for that https://github.com/kubernetes-sigs/external-dns/blob/4b2fd495d944f9a67cdeb5260db95c5f2367d8bf/internal/testutils/log.go#L36

Reply:
Probably i did not get where to add the log test, you can guide me, i added another unit test func however.

Reply:
You could add this function in same test or create a new test

example usage in other tests https://github.com/search?q=repo%3Akubernetes-sigs%2Fexternal-dns%20LogsToBuffer&type=code